src/swarm-tui/Swarm/TUI/Model/StateUpdate.hs

```diff
@@ -127,8 +127,16 @@ initPersistentState ::
   m (RuntimeState, UIState, KeyEventHandlingState)
 initPersistentState opts@(AppOpts {..}) = do
   (warnings :: Seq SystemFailure, (initRS, initUI, initKs)) <- runAccum mempty $ do
-    rs <- initRuntimeState $ RuntimeOptions pausedAtStart (Set.member LoadTestingScenarios debugOptions)
-    ui <- initUIState speed (not (skipMenu opts)) debugOptions
+    rs <-
+      initRuntimeState
+        RuntimeOptions
+          { startPaused = pausedAtStart
+          , pauseOnObjectiveCompletion = autoShowObjectives
+          , loadTestScenarios = Set.member LoadTestingScenarios debugOptions
+          }
+    let showMainMenu = not (skipMenu opts)
+    ui <- initUIState UIInitOptions {..}
+    -- \$ speed (not (skipMenu opts)) debugOptions
```

Comment:
Seems like stray dead code --- should this be removed?

Reply:
Yes, these were the original parameters passed to `initUIState`. Now, only the `showMainMenu` needs to be declared, and the others flow from one record to another.